### PR TITLE
Migrate Viz to use Pydantic V2

### LINKED ISSUE #1603: Support for Pydantic V2 [Planned for Q1 2024]
## Description

* This will bring pydantic v2 features to Kedro Viz

## Context
* Pydantic v2 has changed its implementation to use Rust under the hood which makes it faster

## Possible Implementation
https://docs.pydantic.dev/2.0/migration/

Timelines - We will work on the migration to pydantic v2 next year (Q1, 2024)

## Possible Alternatives
NA

## Checklist

- [x] Include labels so that we can categorise your feature request

### PR DESCRIPTION
**Pull request by ravi-kumar-pilla**
## Description

Resolves #1603 

## Development notes

* Followed [pydantic migration guide](https://docs.pydantic.dev/latest/migration/ )
* Partially updated code using `bump-pydantic` tool
* Manually updated `@validators` to `@field_validators`
* Updated requirements.txt with new pydantic version
* Updated test files as per lint suggestions

## QA notes

* All pytests should pass and `kedro viz run` should work as expected

**Note**: Observed flowchart node arrangements change (mirrored) due to the migration, though the data remains un-affected as below -

**Before Migration:**
 
current_placement

**After Migration:**
pydantic_v2_placement

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by astrojuanlu** [comment]
It's happeninggggggggg

**Comment by MarkusSagen** [comment]
Fantastic!!

File: package/requirements.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@ packaging~=23.0
 kedro>=0.18.0
 ipython>=7.0.0, <9.0
 fastapi>=0.73.0,<0.200.0
-pydantic<2
+pydantic>=2
```
Comment (rashidakanchwala) [inline package/requirements.txt:5]: Do we need to put this, or should we just bump the minimum version of fastAPI ?

File: package/features/steps/lower_requirements.txt [hunk 2]
```diff
@@ -13,4 +13,3 @@ strawberry-graphql==0.192.0
 networkx==2.5
 orjson==3.9
 secure==0.3.0
-pydantic==1.10
```
Comment (noklam) [inline package/features/steps/lower_requirements.txt:16]: This seems different from the discussion in https://github.com/kedro-org/kedro-viz/issues/1603. Just to confirm, we are dropping Pydantic v1 support altogether now.

Comment (ravi-kumar-pilla) [reply package/features/steps/lower_requirements.txt:16]: yes we are dropping support for pydantic v1 as there are issues with the v1 api mentioned [here](https://github.com/tiangolo/fastapi/issues/10360) and an open PR - https://github.com/tiangolo/fastapi/pull/10223/files

**Review by rashidakanchwala — APPROVED** [review]
great work ! thanks @ravi-kumar-pilla

**Pull request merged**